lms/djangoapps/course_api/blocks/transformers/student_view.py

```diff
@@ -72,6 +74,23 @@ def transform(self, usage_info, block_structure):
         """
         Mutates block_structure based on the given usage_info.
         """
+
         for block_key in block_structure.post_order_traversal():
             if block_structure.get_xblock_field(block_key, 'category') not in self.requested_student_view_data:
                 block_structure.remove_transformer_block_field(block_key, self, self.STUDENT_VIEW_DATA)
+            student_view_data = block_structure.get_transformer_block_field(
+                block_key,
+                self,
+                self.STUDENT_VIEW_DATA,
+            )
+
+            if student_view_data and usage_info.user and 'html' in student_view_data:
```

Comment:
Rather than checking for `'html' in student_view_data`, can you change it to: `block_key.block_type == 'html'`?   Just in case another block-type's `student_view_data` happens to contain `html`.

Reply:
Addressed by https://github.com/edx/edx-platform/pull/15905/commits/86235d261e710f16a4790a761e21b604e4883342#diff-1a30c64335d7765247aec974a4d8601eR88